lib/miq_expression.rb

```diff
@@ -1623,27 +1630,34 @@ def self.ruby_for_date_compare(col_ruby, col_type, tz, op1, val1, op2 = nil, val
 
   def to_arel(exp, tz)
     operator = exp.keys.first
-
     field = Field.parse(exp[operator]["field"]) if exp[operator].kind_of?(Hash) && exp[operator]["field"]
+    if ALLOWED_OPERATORS_EXPRESSION_AS_VALUE.include?(operator) && Field.valid_field?(exp[operator]["value"])
+      value_field = Field.parse(exp[operator]["value"]).arel_attribute if exp[operator].kind_of?(Hash) && exp[operator]["value"]
+    elsif ALLOWED_OPERATORS_EXPRESSION_AS_VALUE.include?(operator)
+      value_field = exp[operator]["value"]
+    end
+
     case operator.downcase
-    when "equal", "="
+    when "equal"
```

Comment:
would prefer keeping these together.

Above changes will hopefully allow you to use `value_field` here